codalab/lib/bundle_cli.py

```diff
@@ -1849,6 +1849,9 @@ def do_rm_command(self, args):
             '  search .limit=<limit>                  : Limit the number of results to the top <limit> (e.g., 50).',
             '  search .offset=<offset>                : Return results starting at <offset>.',
             '',
+            '  search .before=<datetime>              : Returns bundles created before given ISO 8601 timestamp (exclusive).',
```

Comment:
I feel like exclusive is a bit unintuitive...can we just do inclusive?  Might want to also add '.date=<date>'

Reply:
Sure I can add date= and make all inclusive.

Reply:
After thinking about it more I'm actually not sure if inclusive is more intuitive since if I were to say before halloween you would probably assume October 30th right?

Reply:
> By that logic, after Halloween would be Nov. 1. Actually, I think we should allow: created=2019-04-05 created>=2019-04-05 created<=2019-04-05 created>2019-04-05 created<2019-04-05 The only downside is that people will have to quote '<' and '>'. Later we could extend these operators to other fields too.
> […](#)
> On Thu, Feb 20, 2020 at 9:57 PM armantajback ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In codalab/lib/bundle_cli.py <[#2004 (comment)](https://github.com/codalab/codalab-worksheets/pull/2004#discussion_r382409636)> : > @@ -1849,6 +1849,9 @@ def do_rm_command(self, args): ' search .limit=<limit> : Limit the number of results to the top <limit> (e.g., 50).', ' search .offset=<offset> : Return results starting at <offset>.', '', + ' search .before=<datetime> : Returns bundles created before given ISO 8601 timestamp (exclusive).', After thinking about it more I'm actually not sure if inclusive is more intuitive since if I were to say before halloween you would probably assume October 30th right? — You are receiving this because you commented. Reply to this email directly, view it on GitHub <#2004?email_source=notifications&email_token=AADKJODDKWKYPSE6KEGAVVTRD5UOLA5CNFSM4KW47VJKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCWM74EI#discussion_r382409636>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AADKJOF2SDA4H4LT27OTGB3RD5UOLANCNFSM4KW47VJA> .

I think that makes sense. Should we allow hours/minutes/seconds precision? Or support dates only?

Reply:
> Ideally all datetimes, including dates. On Thu, Feb 20, 2020 at 10:11 PM armantajback <notifications@github.com> wrote:
> […](#)
 Right now search_bundles is splitting based on what comes before any =. It feels a little weird to have to change the way we are parsing and then change the queries behavior based on what we split on. I also think the quoting will be unintuitive. I would rather support
.before
.before_or_on
.on
.after
.after_or_on
Thoughts?

Reply:
I do have some concerns around behavior for *on functionality. When users provide a time and we want to check for inclusion or a match we have no way of telling whether a time was provided without parsing the provided date ourselves. For example if they want to check for bundles on or before 2019-09-02T00:02, we might want to return every bundle before 00:02:59. No parser I could find tells us whether the seconds field is specified.  
At this point I think we should keep things simpler and let users write queries and manipulate times themselves with before and after to end up with "date=" or "on=" behavior.

Reply:
I'm fine with only having `before` and `after` work on datetimes (not try to do something with dates).  I think they should both be inclusive or exclusive.  Inclusive seems easier to think about (even though it's not the literal interpretation)?

Reply:
Since users can manipulate times and dates couldn't they get inclusivity for free without us having to parse dates ourselves though?

Reply:
All I meant by inclusive is use `<=` and `>=` instead of `<` and `>`.